### PR TITLE
Add support to download CSI json from object store

### PR DESCRIPTION
**Pull request by anshulahuja98**
Thank you for contributing to Velero!

# Please add a summary of your change
Expose -csi-volumesnapshotcontents.json.gz and -csi-volumesnapshots.json.gz files through the DownloadRequest CRs.

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/apis/velero/v1/download_request_types.go [hunk 1]
```diff
@@ -29,13 +29,15 @@ type DownloadRequestSpec struct {
 type DownloadTargetKind string
```
Comment (sseago) [inline pkg/apis/velero/v1/download_request_types.go:29]: The `+kubebuilder:validation:Enum` line above needs the new entries added. Did you manually add those to the downloadrequest CRD file? CRDs are generated via `make update`, so anything added manually will be removed on next update.

**Review by sseago — CHANGES_REQUESTED** [review]

File: pkg/apis/velero/v1/download_request_types.go (hunk 1 above)
Comment (anshulahuja98) [reply pkg/apis/velero/v1/download_request_types.go:29]: I had manually edited it earlier, incorporated your comment and added it to the annotation. Ran the autogen

Let me know if any other concerns with this PR

I will also add test cases if rest of the things look good to you.

**Comment by anshulahuja98** [comment]
will resolve the conflicts

**Comment by anshulahuja98** [comment]
@sseago can you pl review again.

**Comment by blackpiglet** [comment]
@anshulahuja98 
Hi, the CI check action failed with `CRD verification - failed! Generated CRDs are out-of-date, please run 'make update' and 'git add' the generated file(s).`
Please run `make update` locally and push the generated code to resolve the error.

**Comment by anshulahuja98** [comment]
> @anshulahuja98 Hi, the CI check action failed with `CRD verification - failed! Generated CRDs are out-of-date, please run 'make update' and 'git add' the generated file(s).` Please run `make update` locally and push the generated code to resolve the error.

Sorry for the inconvenience, I'll check this again.

**Review by blackpiglet — APPROVED** [review]

**Comment by anshulahuja98** [comment]
@blackpiglet  / @sseago 
Hey can you help me understand why this did not go out in the 1.9.1 release?
How can I make sure that this goes out in the next release.

**Comment by blackpiglet** [comment]
@anshulahuja98 
By default, the code change in main will not be back-port to previous release's minor release.
If the change is needed, please notice Velero team before release.
I'm sure this change will be included in v1.10.

**Comment by anshulahuja98** [comment]
Thank you for explaining this @blackpiglet 

How can stay up to date on any upcoming releases?
I see that the github project board is not up to date.

Basically, if I want to see that a new minor version 1.9.2 is going out. If yes then when and then I can try to engage the velero team to include it.

OR when a major release will go out 1.10?

**Comment by blackpiglet** [comment]
@anshulahuja98 
By far, Velero team also have no clear planning of Velero v1.9.2. 
We have an internal planning document of v1.10, which is updated in recent community meeting.
If you want to have the issue in next minor release, just add comments in the issue, and Velero team will dicuss. If there is no big impat of functionality, there is a big change the PR will be cherry-picked.

**Pull request merged**